Frameworks/CoreText/DWriteWrapper.mm

```diff
@@ -0,0 +1,395 @@
+    static ComPtr<IDWriteFactory> s_dwriteFactory = CreateDWriteFactoryInstance();
+    return s_dwriteFactory;
+}
+
+/**
+ * Helper method to return the user set default locale string.
+ *
+ * @return use set locale string as wstring.
+ */
+static wstring _getUserDefaultLocaleName() {
+    wchar_t localeName[LOCALE_NAME_MAX_LENGTH];
+    int defaultLocaleSuccess = GetUserDefaultLocaleName(localeName, LOCALE_NAME_MAX_LENGTH);
+
+    // If the default locale is returned, find that locale name, otherwise use "en-us".
+    if (defaultLocaleSuccess) {
+        return wstring(localeName);
+    } else {
+        return wstring(L"en-us");
+    }
+}
+
+/**
+ * Helper method to convert IDWriteLocalizedStrings object to NSString object.
+ *
+ * @parameter localizedString IDWriteLocalizedStrings object to convert.
+ *
+ * @return NSString object.
+ */
+static NSString* _convertLocalizedStringToNSString(IDWriteLocalizedStrings* localizedString) {
+    if (localizedString == NULL) {
+        TraceError(TAG, L"The input parameter is invalid!");
+        return nil;
+    }
+
+    // Get the default locale for this user.
+    wstring localeName = _getUserDefaultLocaleName();
+
+    uint32_t index = 0;
+    BOOL exists = false;
+
+    // If the default locale is returned, find that locale name, otherwise use "en-us".
+    THROW_IF_FAILED(localizedString->FindLocaleName(localeName.c_str(), &index, &exists));
+    if (!exists) {
+        THROW_IF_FAILED(localizedString->FindLocaleName(L"en-us", &index, &exists));
+    }
+
+    // If the specified locale doesn't exist, select the first on the list.
+    if (!exists) {
+        index = 0;
+    }
+
+    // Get the string length.
+    uint32_t length = 0;
+    THROW_IF_FAILED(localizedString->GetStringLength(index, &length));
+
+    // Get the string.
+    auto wcharString = make_unique<wchar_t[]>(length + 1);
+    THROW_IF_FAILED(localizedString->GetString(index, &wcharString[0], length + 1));
+
+    return [[[NSString alloc] initWithBytesNoCopy:&wcharString[0] length:length + 1 encoding:NSUTF16StringEncoding freeWhenDone:YES]
+        autorelease];
+}
+
+/**
+ * Helper method to create a IDWriteTextFormat object given _CTTypesetter object and string range.
+ *
+ * @parameter ts _CTTypesetter object.
+ * @parameter range string range to consider for rendering.
+ *
+ * @return the created IDWriteTextFormat object.
+ */
+static ComPtr<IDWriteTextFormat> _createDWriteTextFormat(_CTTypesetter* ts, CFRange range) {
+    // Get the direct write factory instance
+    ComPtr<IDWriteFactory> dwriteFactory = GetDWriteFactoryInstance();
+
+    // TODO::
+    // Get font family name details so that can be used here. For now this is hardcoded to "Gabriola".
+
+    // Note here we only look at attribute value at first index of the specified range as we can get a default faont size to use here.
+    // Per string range attribute handling will be done in _createDWriteTextLayout.
+    NSRange curRange = NSMakeRangeFromCF(range);
```

Comment:
bit silly here - you remade the CFRange into an NSRange just to look at its location